cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -398,6 +403,110 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 	return nil
 }
 
+func (a *StaticAutoscaler) handleQuotaAndStockoutErrors(currentTime time.Time) {
+	currentInstancesPerNodeGroup := a.clusterStateRegistry.GetCloudProviderNodeInstances()
+	previousInstancesPerNodeGroup := a.clusterStateRegistry.GetPreviousCloudProviderNodeInstances()
+	nodeGroups := a.CloudProvider.NodeGroups()
+
+	for _, nodeGroup := range nodeGroups {
+		a.handleQuotaAndStockoutErrorsForNodeGroup(
+			nodeGroup,
+			currentInstancesPerNodeGroup[nodeGroup.Id()],
+			previousInstancesPerNodeGroup[nodeGroup.Id()],
+			currentTime)
+	}
+}
+
+func (a *StaticAutoscaler) handleQuotaAndStockoutErrorsForNodeGroup(
+	nodeGroup cloudprovider.NodeGroup,
+	currentInstances []cloudprovider.Instance,
+	previousInstances []cloudprovider.Instance,
+	currentTime time.Time) {
+
+	_, currentErrorCodeToInstance := a.buildInstanceToOutOfResourcesErrorCodeMappings(currentInstances)
+	previousInstanceToErrorCode, _ := a.buildInstanceToOutOfResourcesErrorCodeMappings(previousInstances)
```

Comment:
You never actually use the mapping. All you care about is the key set.

Reply:
Yeah noticed that. I could return:
* map EC->Instance
* map Instance->bool instead

Or have two methods to compute those independenetly.
That will be a bit more code though.
Let me know if you really want to change it.